src/java/org/apache/cassandra/index/sai/disk/v1/bbtree/BlockBalancedTreeReader.java

```diff
@@ -0,0 +1,455 @@
+            {
+                long pointer = postingsIndex.getPostingsFilePointer(nodeID);
+                postingLists.add(initFilteringPostingReader(pointer, fixedBitSet));
+            }
+        }
+
+        void visitNode(PriorityQueue<PeekablePostingList> postingLists, byte[] cellMinPacked, byte[] cellMaxPacked) throws IOException
+        {
+            byte[] splitPackedValue = index.getSplitPackedValue();
+            BytesRef splitDimValue = index.getSplitDimValue();
+            assert splitDimValue.length == bytesPerValue;
+
+            // make sure cellMin <= splitValue <= cellMax:
+            assert FutureArrays.compareUnsigned(cellMinPacked, 0, bytesPerValue, splitDimValue.bytes, splitDimValue.offset, splitDimValue.offset + bytesPerValue) <= 0 : "bytesPerValue=" + bytesPerValue;
+            assert FutureArrays.compareUnsigned(cellMaxPacked, 0, bytesPerValue, splitDimValue.bytes, splitDimValue.offset, splitDimValue.offset + bytesPerValue) >= 0 : "bytesPerValue=" + bytesPerValue;
+
+            // Recurse on left subtree:
+            System.arraycopy(cellMaxPacked, 0, splitPackedValue, 0, packedBytesLength);
+            System.arraycopy(splitDimValue.bytes, splitDimValue.offset, splitPackedValue, 0, bytesPerValue);
+
+            index.pushLeft();
+            collectPostingLists(postingLists, cellMinPacked, splitPackedValue);
+            index.pop();
+
+            // Restore the split dim value since it may have been overwritten while recursing:
+            System.arraycopy(splitPackedValue, 0, splitDimValue.bytes, splitDimValue.offset, bytesPerValue);
+            // Recurse on right subtree:
+            System.arraycopy(cellMinPacked, 0, splitPackedValue, 0, packedBytesLength);
+            System.arraycopy(splitDimValue.bytes, splitDimValue.offset, splitPackedValue, 0, bytesPerValue);
+            index.pushRight();
+            collectPostingLists(postingLists, splitPackedValue, cellMaxPacked);
+            index.pop();
+        }
+
+        @SuppressWarnings({"resource", "RedundantSuppression"})
+        private PeekablePostingList initFilteringPostingReader(long offset, FixedBitSet filter) throws IOException
+        {
+            final PostingsReader.BlocksSummary summary = new PostingsReader.BlocksSummary(postingsSummaryInput, offset);
+            PostingsReader postingsReader = new PostingsReader(postingsInput, summary, listener.postingListEventListener());
+            return PeekablePostingList.makePeekable(new FilteringPostingList(filter, postingsReader));
+        }
+
+        private FixedBitSet visitDocValues(IndexInput in, int count, IntersectVisitor visitor, short[] origIndex) throws IOException
+        {
+            int commonPrefixLength = readCommonPrefixLength(in);
+            int compressedDimension = readCompressedDimension(in);
+            // If the compressed dimension is -1 it means that all values in the block are equal
+            if (compressedDimension == -1)
```

Comment:
...in other words, could this just be: `if (commonPrefixLength == bytesPerValue)` ?

Reply:
And, ignoring the above comment, this is correct as well. I'll get rid of the flag entirely and use that as a test.